controllers/remote/cluster_cache.go

```diff
@@ -0,0 +1,203 @@
+/*
```

Comment:
Would it make sense to move the bulk of the implementation here to a library so that it would be easier to consume outside of the core controller manager? I wouldn't expect any providers (especially out of tree) to import anything under `controllers`

Reply:
This is actually a library. I've (privately) questioned why the `remote` package is in `controllers/remote` and not somewhere else. I'd say we should file an issue to move `remote` out of `controllers` for 0.4.x. WDYT?

Reply:
i initially put it under there because it was code to be used within capi controllers, it expanded over time though